packages/container/lib/container.js

```diff
@@ -330,8 +330,7 @@ function lookup(container, fullName, options = {}) {
 }
 
 function isSingletonClass(container, fullName, { instantiate, singleton }) {
-  return (singleton !== false && isSingleton(container, fullName)) &&
-         (instantiate === false && !shouldInstantiate(container, fullName));
+  return singleton !== false && isSingleton(container, fullName) && !instantiate && !shouldInstantiate(container, fullName);
 }
 
 function isSingletonInstance(container, fullName, { instantiate, singleton }) {
```

Comment:
We typically want to avoid lines that are too long. I also feel the former is easier to read